src/components/ui/members/member-list.tsx

```diff
@@ -1,12 +1,27 @@
 import { Card, CardContent } from "@/components/ui/card";
 import { User } from "@/types/user";
 import { MemberCard } from "./member-card";
+import { CoachingRelationshipWithUserNames } from "@/types/coaching_relationship_with_user_names";
 
 interface MemberListProps {
   users: User[];
+  relationships: CoachingRelationshipWithUserNames[];
+  onRefresh: () => void;
 }
 
-export function MemberList({ users }: MemberListProps) {
+export function MemberList({
+  users,
+  relationships,
+  onRefresh,
+}: MemberListProps) {
+  // Create a mapping of user IDs to their associated relationships
+  const userRelationshipsMap = users.reduce((acc, user) => {
+    acc[user.id] = relationships.filter(
+      (rel) => rel.coach_id === user.id || rel.coachee_id === user.id
+    );
+    return acc;
+  }, {} as Record<string, CoachingRelationshipWithUserNames[]>);
```

Comment:
```suggestion
  }, {} as Record<Id, CoachingRelationshipWithUserNames[]>);
```

I think the first type is supposed to hold a Member's Id, but I'm not entirely sure. But if this is true, `Id` type is an appropriate change.

Reply:
accebcf99c476dcd909eb3b27c415223ebecd387